### ISSUE TITLE
adjtimex reports STA_UNSYNC after ntpd-rs restart

### ISSUE THREAD

**Issue by mcpherrinm** [pendulum-project/ntpd-rs#1553]
We monitor our instances with the Prometheus Node Exporter, which exports a metric called `node_timex_sync_status`. It reports if `clock synchronized to a reliable server`.  Looking at our metrics, it seems like it started reporting synced a few days after we'd restarted ntpd-rs.

This is exporting the result of calling the Go function `unix.Adjtimex`, which calls the C function `adjtimex`.  I've included two small reproducer programs below to avoid including the whole prometheus stack in the description.

I'm not deeply familiar with how the Linux APIs work here, but other time keeping daemons (chrony and systemd-timesyncd) seem to set this flag.  I don't know if that's done implicitly or explicilty.

The following roughly-equivalent programs print 0 when running chrony and the clock is synced, but don't with ntpdrs:

```
#include <stdio.h>
#include <sys/timex.h>

int main() {
  struct timex t = {0};
  int status = adjtimex(&t);
  printf("%d\n", t.status & STA_UNSYNC);
}
```
```
package main

import "golang.org/x/sys/unix"
import "fmt"

func main() {
	var timex = new(unix.Timex)
	status, err := unix.Adjtimex(timex)
	fmt.Println(status, err)
}
```

**Comment by davidv1992** [comment]
This turned out to be a subtle bug in how we interacted with the unix adjtimex api, which should be fixed in 1.2.1

**Comment by mcpherrinm** [comment]
I've confirmed this issue is resolved in our deployment
